pkg/cloudprovider/aws/amifamily/bootstrap/eksbootstrap.go

```diff
@@ -16,15 +16,23 @@ package bootstrap
 
 import (
 	"bytes"
+	"context"
 	"encoding/base64"
 	"fmt"
+	"net"
 	"strings"
 	"sync"
 
+	"knative.dev/pkg/logging"
+
 	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
 )
 
 type EKS struct {
+	context.Context
```

Comment:
I don't think this is the right place to move these to.  I think the way you had it before was okay, but the parameter that the bootstrap script cares about is something denoting if the node should be an ipv6 node or not. The bootstrap script doesn't need to care about how that was decided (it could be a user setting it explicitly to be ipv6 or the mechanism to discover the kube-dns IP and check if that is ipv4 or ipv6). 

EKSBootstrap doesn't bootstrap bottlerocket which should also support IPv6. 

I would suggest going back to the way you had it before, but instead of passing a second DNS IP, to add an `IPFamily` field to the base Bootstrap Options. That way the eks bootstrapper and the bottlerocket bootstrapper can work with that parameter directly.  You can also add a parameter for the `IPv6CIDR`

Reply:
Thanks, I added the IPv6CIDR params in the struct.

On the other hand, I found the users need to pass the cluster dns ip explicitly to launch bottlerocket nodes correctly when their cluster cannot use EKS API . 
https://github.com/bottlerocket-os/bottlerocket/blob/5d6cc099d247ecc12e8250898c48d9ee762f8c42/sources/api/pluto/src/main.rs#L256

I think we have to pass cluster dns IP to the bottlerocket bootstrap code in karpenter which generates the toml for bottlerocket even if the user doesn't specify their DNS IP.

Could I post another issue for the bottlerocket?